scripts/linux/agent_diagnostics.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+# agent_diagnostics.sh collects all artifacts required for troubleshooting,
+# including logs, configuration files, and application files. Those artifacts
+# are compressed into a zip file, along with an output file of relevant information
+# about the system for quick referrence and an inventory of the files included
+# in the zip file.
+
+if [[ "${UID}" != 0 ]]; then
+  (>&2 echo "Error:  $0 must be run as root")
+  exit 1
+fi
+
+# Some global variables
+JCPATH="/opt/jc"
+JCLOG="/var/log/"
+STAMP=$( date +"%Y%m%d%H%M%S" )
+ZIPFILE="./jc${STAMP}.zip"
+declare -a INVENTORY
+
+# Is jcagent installed? If not, exit.
+if [[ ! -d "${JCPATH}" ]]; then
+  echo "JCAGENT IS NOT INSTALLED ON THIS MACHINE."
+  exit 1
+fi
+
+# Is zip installed?
+if ! which zip 2> /dev/null; then
+  ZPATH="false"
+else
+  ZPATH=$(which zip)
+fi
+
+function indent() {
+  # Formatting for output.log
+  sed 's/^/\t\t/g'
+}
+
+function zipjc() {
+  # Take inventory of files to be zipped
+  for i in "${JCPATH}"/*; do
+    INVENTORY+=("${i}")
+  done
+
+  # check to see if zip exists.
+  if [[ "$ZPATH" = "false" ]]; then
+    ZIPIT="zip is not installed. please send the following files with your support request:"
+  else
+    if [[ -f "${ZIPFILE}" ]]; then
+      mv "${ZIPFILE}" ./jc"${STAMP}".bak.zip
+      zip -r "*.crt" -r "${ZIPFILE}" "${JCPATH}" > /dev/null 1
```

Comment:
Furthermore, we should exclude the `.key` files as we do in the Windows script.

Reply:
I've moved the exclusions up to the collection of the inventory list, instead of trying to do it inline in the zip command, which seems like a cleaner solution.

currently, I'm checking for the presence of zip, and if it doesn't exist, I simply make note it in the output. I am planning to change this to fall back to tar on systems that don't have zip (tar is a requirement for supported linux systems), but for now will hint to install zip or include the files in the inventory list with the support request.